pkg/build/api/types.go

```diff
@@ -389,6 +389,8 @@ type BuildTriggerPolicy struct {
 // BuildTriggerType refers to a specific BuildTriggerPolicy implementation.
 type BuildTriggerType string
 
+//NOTE: Adding a new trigger type requires changing the dropUnknownTriggers function in
+// github.com/openshift/origin/pkg/build/regisry/strategy.go
```

Comment:
if we build a list of valid types here, we can use it in dropTriggers... keeps reference local
```
var KnownTriggerTypes = util.NewStringSet(GitHubWebHookBuildTriggerType, GenericWebHookBuildTriggerType, ImageChangeBuildTriggerType, ConfigChangeBuildTriggerType)
```